### PR TITLE
Throttling test cases and bug fixes

### LINKED ISSUE #1839: If jwt claim based throttling is enabled and a certain resource's auth is disabled, results in NPE
### Description:
Null Pointer exception happens if the $subject is met.

### Steps to reproduce:
Jwt Claim Based throttling is applied.
(Then the advanced policy is changed to a generic count-based policy again.)
One resource's authentication is disabled.
Try to invoke the resource.

### Affected Product Version:
m10-SNAPSHOT

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s): Docker with APIM

---

### LINKED ISSUE #1898: NPE when jwt claim conditions are enabled without backend token generation
### Description:

Null pointer exception occurs when jwt claim condition based throttling is enabled but backend token generation is disabled.
```
SEVERE: Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed@638e4c7d
java.lang.NullPointerException
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleDataHolder.isJwtClaimPresent(ThrottleDataHolder.java:513)
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleDataHolder.isThrottledByCondition(ThrottleDataHolder.java:367)
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleDataHolder.isAdvancedThrottled(ThrottleDataHolder.java:287)
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleFilter.checkResourceThrottled(ThrottleFilter.java:229)
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleFilter.doThrottle(ThrottleFilter.java:136)
	at org.wso2.choreo.connect.enforcer.throttle.ThrottleFilter.handleRequest(ThrottleFilter.java:67)
	at org.wso2.choreo.connect.enforcer.api.API.executeFilterChain(API.java:41)
	at org.wso2.choreo.connect.enforcer.api.RestAPI.process(RestAPI.java:98)
	at org.wso2.choreo.connect.enforcer.server.RequestHandler.process(RequestHandler.java:60)
	at org.wso2.choreo.connect.enforcer.grpc.ExtAuthService.check(ExtAuthService.java:50)
	at io.envoyproxy.envoy.service.auth.v3.AuthorizationGrpc$MethodHandlers.invoke(AuthorizationGrpc.java:245)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:180)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:814)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```
### Steps to reproduce:
- enable throttling
- enable jwt claim condition based throttling
- create jwt claim condition based advance throttling policy
- create an API and assign above policy
- invoke API until default limit hits and observer the behavior.
### Affected Product Version:

0.9.0-alpha
### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s): Docker with apim

---
#### Related Issues:

#1839

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

`ThrottlingBaseTestCase` includes some of the util methods used by other testcases. All other throttle test cases should extend this class.
Possible improvements:
- Test with disabled throttle condition configurations (requires restart)

Includes:
- Reorg some of the apim test util classes/methods
- Test cases for global throttling.
- Bug fixes
### Issues

Fixes #1898
Fixes #1839

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Mac

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/adapter/adapter.go [hunk 1]
```diff
@@ -221,13 +221,10 @@ func Run(conf *config.Config) {
 		go synchronizer.UpdateRevokedTokens()
 		// Fetch Key Managers from APIM
 		synchronizer.FetchKeyManagersOnStartUp(conf)
-	}
-
-	throttlingEnabled := conf.Enforcer.Throttling.EnableGlobalEventPublishing
-	if throttlingEnabled {
```
Comment (Rajith90) [inline adapter/internal/adapter/adapter.go:227]: Is there a reason to remove this check

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java [hunk 2]
```diff
@@ -63,6 +63,8 @@
 
 public class APIMLifecycleBaseTest extends APIMWithMgwBaseTest {
     private static final Logger LOGGER = LoggerFactory.getLogger(APIMLifecycleBaseTest.class);
+    protected static final String SAMPLE_API_NAME = "PetStore";
+    protected static final String SAMPLE_API_CONTEXT = "petstore";
```
Comment (Rajith90) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java:67]: Usually we don't keep API names like this in the base class

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/APIManagerWithMgwServerInstance.java [hunk 3]
```diff
@@ -70,14 +73,29 @@ public APIManagerWithMgwServerInstance(String confPath, boolean tlsEnabled) thro
                     mgwTmpServerPath + File.separator + "docker-compose" + File.separator + "choreo-connect-with-apim"
                             + File.separator + "conf" + File.separator + "config.toml");
         }
+        Logger enforcerLogger = LoggerFactory.getLogger("Enforcer");
+        Logger adapterLogger = LoggerFactory.getLogger("Adapter");
+        Logger routerLogger = LoggerFactory.getLogger("Router");
+        Slf4jLogConsumer enforcerLogConsumer = new Slf4jLogConsumer(enforcerLogger);
+        Slf4jLogConsumer adapterLogConsumer = new Slf4jLogConsumer(adapterLogger);
+        Slf4jLogConsumer routerLogConsumer = new Slf4jLogConsumer(routerLogger);
```
Comment (Rajith90) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/APIManagerWithMgwServerInstance.java:81]: Shall we add APIM logs as well

File: adapter/internal/adapter/adapter.go (hunk 1 above)
Comment (praminda) [reply adapter/internal/adapter/adapter.go:227]: this is adapter side check. here we're checking if throttling is enabled at enforcer config to fetch throttle data to adapter from CP. Ideally we should fetch eventhub data to adapter if eventhub is enabled and enforcers can subscribe to those data via xds if throttling is enabled

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java (hunk 2 above)
Comment (praminda) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java:67]: issue is with the variable value or the class its placed in?

Comment (Rajith90) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java:67]: Lets move the API names to the static section of the class

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/APIManagerWithMgwServerInstance.java (hunk 3 above)
Comment (praminda) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/context/APIManagerWithMgwServerInstance.java:97]: Done with ac25b0a

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java (hunk 2 above)
Comment (praminda) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/apim/APIMLifecycleBaseTest.java]: Moved to testconstants f32cf4d

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**